be/src/util/string_parser.hpp

```diff
@@ -624,27 +624,29 @@ T StringParser::string_to_decimal(const char* s, int len, int type_precision, in
     }
 
     int precision = 0;
+    int max_digit = type_precision - type_scale;
     bool found_exponent = false;
     int8_t exponent = 0;
     T value = 0;
     for (int i = 0; i < len; ++i) {
         const char& c = s[i];
-        if (LIKELY('0' <= c && c <= '9')) {
+        if (found_dot && max_digit < (precision - scale)) {
```

Comment:
why `found_dot` here? maybe `!found_dot` here is right. And maybe place this judgement after precisions' increments rather than the front of loop body will be more efficient.

Reply:
make sense!